_aicspylibczi/ImageFactory.h

```diff
@@ -11,19 +12,34 @@ namespace pylibczi {
       using PixelType = libCZI::PixelType;
       using CtorMap = std::map<libCZI::PixelType,
                                std::function<std::shared_ptr<Image>(
-                                   std::vector<size_t>, libCZI::PixelType pixel_type_, const libCZI::CDimCoordinate* plane_coordinate_,
-                                   libCZI::IntRect box_, int mIndex_)
+                                   std::vector<size_t>, libCZI::PixelType pixel_type_,
+                                   const libCZI::CDimCoordinate* plane_coordinate_,
+                                   libCZI::IntRect box_, ImagesContainerBase* bptr,
+                                   size_t img_index_, int mIndex_)
                                > >;
       using SplitCtorMap = std::map<libCZI::PixelType, std::function<std::shared_ptr<Image>( std::shared_ptr<Image> img_, int channel_) > >;
 
-      static CtorMap s_pixelToImage;
-      static SplitCtorMap s_pixelToSplit;
+      static CtorMap s_pixelToImageConstructor;
+
+      ImagesContainerBase::ImagesContainerBasePtr m_imgContainer;
 
   public:
+      ImageFactory(libCZI::PixelType pixel_type_, size_t pixels_in_all_images_)
+      : m_imgContainer(ImagesContainerBase::getTypedAsBase(pixel_type_, pixels_in_all_images_))
+      {}
+
+      ImagesContainerBase::ImagesContainerBasePtr returnMemory(void){
```

Comment:
is it clear enough to a caller that this function has side effects on the object?  Should it be called something like relinquish or transferMemory?